crates/vault-registry/src/types.rs

```diff
@@ -28,37 +28,54 @@ pub enum Version {
 
 #[derive(Debug, PartialEq)]
 pub enum CurrencySource<T: frame_system::Config> {
-    /// used by vault to back issued tokens
+    /// Used by vault to back issued tokens
     Collateral(<T as frame_system::Config>::AccountId),
     /// Collateral that is locked, but not used to back issued tokens (e.g. griefing collateral)
     Griefing(<T as frame_system::Config>::AccountId),
     /// Unlocked balance
     FreeBalance(<T as frame_system::Config>::AccountId),
-    /// funds within the liquidation vault
+    /// Locked balance (like collateral but doesn't slash)
+    ReservedBalance(<T as frame_system::Config>::AccountId),
+    /// Funds within the liquidation vault
     LiquidationVault,
 }
 
 impl<T: Config> CurrencySource<T> {
     pub fn account_id(&self) -> <T as frame_system::Config>::AccountId {
         match self {
-            CurrencySource::Collateral(x) | CurrencySource::Griefing(x) | CurrencySource::FreeBalance(x) => x.clone(),
+            CurrencySource::Collateral(x)
+            | CurrencySource::Griefing(x)
+            | CurrencySource::FreeBalance(x)
+            | CurrencySource::ReservedBalance(x) => x.clone(),
             CurrencySource::LiquidationVault => Pallet::<T>::liquidation_vault_account_id(),
         }
     }
+
     pub fn current_balance(&self) -> Result<Collateral<T>, DispatchError> {
         match self {
-            CurrencySource::Collateral(x) => Ok(Pallet::<T>::get_rich_vault_from_id(&x)?.data.backing_collateral),
+            CurrencySource::Collateral(x) => {
+                let vault = Pallet::<T>::get_rich_vault_from_id(&x)?;
+                Ok(if vault.data.is_liquidated() {
+                    vault.data.liquidated_collateral
+                } else {
+                    vault.data.backing_collateral
+                })
+            }
             CurrencySource::Griefing(x) => {
-                let backing_collateral = match Pallet::<T>::get_rich_vault_from_id(&x) {
-                    Ok(vault) => vault.data.backing_collateral,
-                    Err(_) => 0u32.into(),
+                let vault = Pallet::<T>::get_rich_vault_from_id(&x)?;
+                let backing_collateral = if vault.data.is_liquidated() {
+                    vault.data.liquidated_collateral + vault.data.backing_collateral
```

Comment:
checked math. Also, we can just always do the addition, since it should be 0 if the vault is not liquidated